rustls/src/quic.rs

```diff
@@ -108,7 +145,62 @@ impl PacketKey {
                 &[],
             )),
             iv: hkdf_expand(secret, IvLen, b"quic iv", &[]),
+            suite,
+        }
+    }
+
+    /// Encrypt a QUIC packet
+    ///
+    /// `buf` contains the header, payload and space for the AEAD tag, with the payload starting at
+    /// `header_len` and ending at `tag_len()` before the end of `buf`.
+    pub fn encrypt(&self, packet: u64, buf: &mut [u8], header_len: usize) -> Result<(), Error> {
+        let (header, payload) = buf.split_at_mut(header_len);
+        let (payload, tag_storage) =
+            payload.split_at_mut(payload.len() - self.key.algorithm().tag_len());
+        let aad = aead::Aad::from(header);
+        let nonce = self.iv.nonce_for(packet);
```

Comment:
Don't we need to also make `Iv` and maybe other parts of the public QUIC API like `nonce_for` private?

Reply:
Ah, good point, done.